addons/pos_razorpay/models/pos_payment_method.py

```diff
@@ -0,0 +1,124 @@
+import logging
+import requests
+
+from odoo.exceptions import UserError
+from odoo import fields, models, api, _
+
+_logger = logging.getLogger(__name__)
+REQUEST_TIMEOUT = 30
+
+class PosPaymentMethod(models.Model):
+    _inherit = 'pos.payment.method'
+
+    razorpay_tid = fields.Char(string='Razorpay Device Serial No', help="Device Serial No \n ex: 7000012300")
+    razorpay_allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('upi', 'UPI'), ('bharatqr', 'BHARATQR')], default='all', help="Choose allow payment mode: \n All/Card/UPI or QR")
+    razorpay_username = fields.Char(string="Razorpay Username", help="Username(Device Login) \n ex: 1234500121")
+    razorpay_api_key = fields.Char(string="Razorpay API Key", help="Used when connecting to Razorpay: https://razorpay.com/docs/payments/dashboard/account-settings/api-keys/")
+    razorpay_test_mode = fields.Boolean(string="Razorpay Test Mode", default=False, help="Turn it on when in Test Mode")
+
+    def _razorpay_get_endpoint(self):
+        if self.razorpay_test_mode:
+            return 'https://demo.ezetap.com/api/3.0/p2padapter/'
+        return 'https://www.ezetap.com/api/3.0/p2padapter/'
+
+    def _get_payment_terminal_selection(self):
+        return super()._get_payment_terminal_selection() + [('razorpay', 'Razorpay')]
+
+    def _call_razorpay(self, endpoint, payload=None):
+        """ Make a request to Razorpay API.
+
+        :param str endpoint: The endpoint to be reached by the request.
+        :param dict payload: The payload of the request.
+        :return The JSON-formatted content of the response.
+        :rtype: dict
+        """
+        endpoint = f"{self._razorpay_get_endpoint()}{endpoint}"
+        with requests.Session() as _session:
+            try:
+                response = _session.post(endpoint, json=payload, timeout=REQUEST_TIMEOUT)
+                response.raise_for_status()
+            except (requests.exceptions.Timeout, requests.exceptions.RequestException) as error:
```

Comment:
requests.exceptions.Timeout is a child of requests.exceptions.RequestException, simply catching RequestException is enough:
https://docs.python-requests.org/en/latest/_modules/requests/exceptions/#:~:text=class%20Timeout(RequestException)%3A

Reply:
Then it should be able to catch `HTTPError`, `ConnectionError`   not a fan of a lot of exceptions at once, just to make sure we do not have bind spot, I wonder we should add `ValueError` incase if we don't get a json response